pkg/idp/oauth/validator.go

```diff
@@ -96,8 +97,14 @@ func (b *IdentityProvider) validateAccessToken(state string, data map[string]int
 			return nil, errors.ErrIdentityProviderOAuthEmailNotFound.WithArgs(b.config.IdentityTokenName)
 		}
 	}
-
 	m := make(map[string]interface{})
+	for k, v := range claims {
+		if _, exists := b.userInfoFields[k]; exists {
```

Comment:
@MartinWallgren , thank you for the contribution. What is the effect of this? Is it possible to only add "select" fields as opposed to everything?